### ISSUE TITLE
Explain Your Issue

### ISSUE THREAD

**Issue by niparis** [wolveix/satisfactory-server#121]
**Describe the Bug**
Server crashes, logs mention some issue with Steam API

selected logs (last lines)

```
[2022.02.03-11.14.03:610][265]LogDTLSHandler: Warning: Empty certificate identifier
[2022.02.03-11.14.03:686][272]LogDTLSHandler: DTLSCertVerifyCallback: Verified
[2022.02.03-11.14.03:694][273]LogDTLSHandler: UpdateHandshake:  Handshaking completed
[2022.02.03-11.14.04:856][411]LogNet: NotifyAcceptingChannel Control 0 server OnlineBeaconHost /Game/FactoryGame/Map/DedicatedserverEntry.DedicatedserverEntry:PersistentLevel.OnlineBeaconHost_2147482544: Accepted
[2022.02.03-11.14.04:856][411]LogNet: Remote platform little endian=1
[2022.02.03-11.14.04:856][411]LogNet: This platform little endian=1
[2022.02.03-11.14.04:856][411]LogBeacon: Beacon Hello
[2022.02.03-11.14.04:923][419]LogBeacon: Client netspeed is 120000
[2022.02.03-11.14.04:923][419]LogOnline: OSS: Creating online subsystem instance for: STEAM
[2022.02.03-11.14.04:923][419]LogOnline: Warning: STEAM: Steam API failed to initialize!
[2022.02.03-11.14.04:923][419]LogOnline: Display: STEAM: OnlineSubsystemSteam::Shutdown()
[2022.02.03-11.14.04:923][419]LogBeacon: Beacon Join FGServerBeaconClient STEAM:(STEAM)076561197997082829
[2022.02.03-11.14.04:923][419]LogReplicationGraph: Warning: No Replication Lists were returned for connection
```

Full logs:
[_satisfactory-server_logs.txt](https://github.com/wolveix/satisfactory-server/files/7994077/_satisfactory-server_logs.txt)

**Your Runtime Command or Docker Compose File**
using the Docker Compose file as in the readme, pointing to my home directory

**System Specs (please complete the following information):**
 - Ubuntu 20.04 on x86_64

**Additional Context**
- download happened fine
- could connect to the server
- crash happened about a minute after the clicking on "create game". I was on the screen to select a biome.

**Comment by wolveix** [comment]
Hey! Thanks for opening this issue.

Hm, that's not an unusual log line, so I doubt it's the root cause. Is the Docker container itself crashing? Or do you just lose your ability to connect to the server from your client? How are you trying to connect to it? Is the server hosted locally or remotely?

**Comment by MikeSonor** [comment]
Same here. Fresh container runs ok until i click "create game" in the client.

**Comment by wolveix** [comment]
@MikeSonor per my previous comment, does the container itself crash?

**Comment by MikeSonor** [comment]
No. Works fine. Crash only appears after i'm trying to create new world (or load a save)
Docker runs on my home server (under vmware VM). Today i'll try to run it in docker desktop (for test).

**Comment by MikeSonor** [comment]
Ok. Here is some results from tests.
Run via bare docker (docker run ...) from my PC - OK. Didn't crash. 
Run via bare docker from server (same one) - OK. Didn't crash
Run via docker-compose from server - OK. Didn't crash

Run via docker stack deploy -c docker-compose.yml  - Same issue. Crash after click on "create world".
(i'm use docker stack for memory and cpu limitation for container)

So it's something wrong with docker stack. I'll try some tweaks later. Maybe write config from scratch.
@niparis do you run it from docker stack as well?

**Comment by wolveix** [comment]
Thanks for the details troubleshooting @MikeSonor! It could be that you've constrained the container too much with the limitations imposed by `stack`, causing world generation/loading to produce a crash

**Comment by MikeSonor** [comment]
huh.. it was 4G of memory before. Now i'm disable memory limit and run docker stats on container from stack. When map was generated memory usage bump to 5.63G. So yeah.. it was memory limitation issue (i guess after few updates satisfactory changed its minimum memory requirement from 4G to 6G maybe)
Now it's works fine)  Thx for push in right direction @wolveix !

**Comment by wolveix** [comment]
@MikeSonor I'm glad to hear it's working for you! The server is still in its early stages, and Coffee Stain Studios are still working on it. You'll want to be fairly lenient on limiting its resources for the time being :)

**Comment by msladek** [comment]
memory footprint looks similar on my end:
```
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
232073      4577  9.6  9.2 8193840 6058312 ?     Ssl  Feb06 610:08 ./Engine/Binaries/Linux/UE4Server-Linux-Shipping FactoryGame
```

**Comment by niparis** [comment]
> Hey! Thanks for opening this issue.
> 
> Hm, that's not an unusual log line, so I doubt it's the root cause. Is the Docker container itself crashing? Or do you just lose your ability to connect to the server from your client? How are you trying to connect to it? Is the server hosted locally or remotely?

1. The server is in my home. It's a dedicated ubuntu machine.
2. My client is on the same LAN, BUT I'm using the external, public facing IP to connect
3. No, the container does not crash
4. The server just appears "offline" from the satisfactory interface

====== extra info ======
Typically when I stop and restart the container it works. So essentially every time I start the container there's a 50% chance it works.

**Comment by niparis** [comment]
> @niparis how much memory on your server?

32GB available. I can't say about usage last time there was an issue.
Right now (similar applications running) there are 21 GB free.

**Comment by wolveix** [comment]
I'm going to close this due to inactivity, but please feel free to re-open this @niparis :)
